internal/sidekick/internal/api/model.go

```diff
@@ -191,13 +191,32 @@ type Service struct {
 	DefaultHost string
 	// The Protobuf package this service belongs to.
 	Package string
+
+	// Synthetic messages for the parameters and body of each request.
+	//
+	// When using discovery docs or OpenAPI:
+	// 1. Not all methods have a request message.
+	// 2. The request message does not include the path and query parameters.
+	// 3. The request message may be shared, so it is risky to modify it.
+	//
+	// Sidekick creates synthetic messages for these specifications. The codecs
+	// need to generate these messages separate from "normal" messages. They
+	// should keep in mind that name clashes are possible.
+	Requests []*Message
```

Comment:
the alternative is to add a `Synthetic: bool` field to an `api.Message`. And then in generation, we could skip generating serde for any message where `m.Synthetic == true`.

And we can skip generating the message at all when `m.Synthetic == true && !m.Parent`. Or something.